### PR TITLE
Check for api reachability after Up'ing cluster in kubetest

### PR DESCRIPTION
**Pull request by shyamjvs**
We should do this to make kubetest more robust.
For eg. https://github.com/kubernetes/kubernetes/issues/46195

/cc @krzyzacy @fejta

### CODE REVIEW

File: kubetest/e2e.go [hunk 1]
```diff
@@ -263,6 +265,22 @@ func run(deploy deployer, o options) error {
...
 
+func checkAPIReachability() error {
+	retries := 5
+	for {
+		_, err := output(exec.Command("./cluster/kubectl.sh", "--match-server-version=false", "version"))
+		if err == nil {
+			return nil
+		}
+		retries--
+		if retries == o {
```
Comment (krzyzacy) [inline kubetest/e2e.go]: s/o/0

Comment (shyamjvs) [inline kubetest/e2e.go]: Done

**Pull request merged**